packages/gatsby-link/src/__tests__/index.tsx

```diff
@@ -8,49 +8,81 @@ import {
   createMemorySource,
   createHistory,
   LocationProvider,
+  NavigateOptions,
 } from "@reach/router"
 import { Link, navigate, withPrefix, withAssetPrefix } from "../"
 
+type IntersectionObserverType = jest.Mock<
+  {
+    observe: (ref: Record<string, unknown>) => void
+    unobserve: (ref: Record<string, unknown>) => void
+    disconnect: () => void
+    trigger: (ref: Record<string, unknown>) => void
+  },
+  [cb: any]
+>
+
+interface ICustomNodeJsGlobal extends NodeJS.Global {
+  __BASE_PATH__: string | undefined
+  __PATH_PREFIX__: string | undefined
+  ___navigate: (
+    to: string,
+    options?: NavigateOptions<Record<string, unknown>>
+  ) => void
+  ___loader: {
+    enqueue: (arg0: string) => Record<string, unknown>
+    hovering?: (arg0: string) => Record<string, unknown>
+  }
+  IntersectionObserver: IntersectionObserverType
+}
+
+declare const global: ICustomNodeJsGlobal
+
 beforeEach(() => {
   global.__BASE_PATH__ = ``
   global.__PATH_PREFIX__ = ``
 })
 
 afterEach(cleanup)
 
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```

Comment:
This PR has a lot of eslint-disable for the explicit function return type. We have that linting rule for a reason 😅 

Please remove all instances of that eslint-disable and add the return types